### DISCUSSION TITLE
Milvus - linux Number of Open Files (ulimit -Hn). Is there a need to increase them for standalone docker compose installation?

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [milvus-io/milvus#34570] [Q&A and General discussion]
Milvus - Linux Number of Open Files (ulimit -Hn). Is there a need to increase them for standalone docker compose installation? 
I once got an error on the old version of milvius?

Is there a need to increase limit on the host system and in docker containers? 

What limit should I set?

**Comment by yhmo** [comment]
All the nodes are running within a standalone(datanode, querynode, indexnode).
Different work nodes do different tasks: 
datanode might trigger compaction task to merge segments, each compaction task needs to read data files parallelly of multiple segments into memory and merge them together. 
querynode might receive load request to load data files of a collection into memory, the load process is also concurrent read.
indexnode builds index for segments, it also reads data files occasionally.

So, it is recommend to set the ulimt to a large value. You can set it to be 10000. If the error still occurs, increase the ulimit value.

**Reply by martinenkoEduard** [reply]
Do I need to set it in the host system?
Or in the docker containers? (in docker-compose file?)

Or Do I need to change this value in the system in the docker compose file?

**Reply by xiaofan-luan** [reply]
usually you don't need to.
I believe we've already changed ulimit inside docker. Unless you hit some too many open file error

**Reply by martinenkoEduard** [reply]
What is the proper way to do it If I ever encounter this error?
If I use docker compose.

Maybe it is wise to increase it even more in further releases.

**Reply by xiaofan-luan** [reply]
ulimit -n 65536

when you hit this problem, usually you are facing some other problems.
Maybe you are just doing too much manual flush or your memory is not enough
